docs/gallery/torontonian_tutorial.ipynb

```diff
@@ -0,0 +1,366 @@
+   "metadata": {},
+   "source": [
+    "We could likewise define it as a proof-of-concept with simple Python code:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 222,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import itertools\n",
+    "import numpy as np\n",
+    "\n",
+    "\n",
+    "def powerset(s):  # empty set handled as special case\n",
+    "  for i in range(1, len(s)+1):\n",
+    "    yield from itertools.combinations(s, i)\n",
+    "\n",
+    "\n",
+    "def matix(mat, idx):\n",
+    "    return mat[idx][:,idx]\n",
+    "\n",
+    "\n",
+    "def torontonian(mat):\n",
+    "    N = len(mat)\n",
+    "    assert((N & 1) == 0)  # N = 2 * d\n",
+    "\n",
+    "    N >>= 1  # determinant of empty matrix is 1\n",
+    "\n",
+    "    res = []\n",
+    "    for Z in powerset(list(range(N))):\n",
+    "        idx = [*Z, *(x + N for x in Z)]\n",
+    "        step = np.eye(len(Z) << 1) - matix(mat, idx)\n",
+    "\n",
+    "        nom = 2 * ((N - len(Z)) & 1 == 0) - 1\n",
+    "        denom = np.sqrt(np.linalg.det(step))\n",
+    "        res.append(nom / denom)\n",
+    "\n",
+    "    term = 2 * ((N & 1) == 0) - 1\n",
+    "    return sum(res) + term"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Using the library"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Import the library in the usual way:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 204,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from thewalrus import tor"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To use it  we need to pass square NumPy arrays as arguments:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 205,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The library provides functions to compute Torontonians of real and complex matrices. The function's arguments must be passed as NumPy arrays."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 226,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "((-16-16j), (-16-16j))"
+      ]
+     },
+     "execution_count": 226,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "size = 10\n",
+    "matrix = 1j * np.ones([size, size])\n",
+    "matrix = np.tril(matrix.conjugate(), -1) + np.triu(matrix, 1) + np.eye(size)\n",
+    "torontonian(matrix), tor(matrix)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As a simple example, the Torontonian of a matrix containing ones on the main diagonal, with its upper triangular containing all values of $1i$ (and thus its lower triangular containing all values of $-1i$) is given by $-1^l(2^{l-1}+2^{l-1}i)$"
```

Comment:
Nice! Did not know about this special case. Could be potentially useful to another test.

Reply:
@nquesada I did not know about it either, I kept experimenting and playing around until I got lucky and found it :)  Lots of NaN results along the way to it.  Basically wanted a simple integer valued example like we have in the other tutorials.  Hope it is useful indeed!